docs/api.md

```diff
@@ -0,0 +1,3 @@
+# Reference
+
+TODO
```

Comment:
maybe include an issue link so that this work is tracked?

only really matters if you arent going to immediately follow up on this

Reply:
This specific page will take the longest as I need to finish the work on the API docs generation library to support restructured text docstrings, but most of them should be a fast follow.